src/main/java/org/example/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -1,27 +1,30 @@
 package org.example.bookstore.repository.book;
 
 import lombok.RequiredArgsConstructor;
-import org.example.bookstore.dto.BookSearchParameters;
+import org.example.bookstore.dto.bookdto.BookSearchParameters;
 import org.example.bookstore.model.Book;
 import org.example.bookstore.repository.SpecificationBuilder;
 import org.example.bookstore.repository.SpecificationProviderManager;
+import org.example.bookstore.repository.book.spec.AuthorSpecificationProvider;
+import org.example.bookstore.repository.book.spec.TitleSpecificationProvider;
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Component;
 
 @RequiredArgsConstructor
 @Component
 public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
 
-    private static final String TITLE = "title";
-    private static final String AUTHOR = "author";
-
+    private final AuthorSpecificationProvider authorSpecificationProvider;
+    private final TitleSpecificationProvider titleSpecificationProvider;
     private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
 
     @Override
     public Specification<Book> build(BookSearchParameters searchParameters) {
         Specification<Book> specification = Specification.where(null);
-        specification = addSpecification(specification, searchParameters.author(), AUTHOR);
-        specification = addSpecification(specification, searchParameters.title(),TITLE);
+        specification = addSpecification(specification, searchParameters.author(),
+                authorSpecificationProvider.getKey());
```

Comment:
You can also make a constant in `authorSpecificationProvider` public and import it here for direct use like a static method instead of calling a method, but that is up to you